dit/dit.py

```diff
@@ -1066,32 +1102,55 @@ def switchback(self, argv):
             self.halt([])
             self.workon([PREVIOUS_FN])
 
-    @command("c", "", SELECT_BY_NAME)
+    @command("c", select=SELECT_BACKWARD)
     def conclude(self, argv):
         self.halt(argv, conclude=True)
 
-    @command("q", "--concluded --verbose --all", SELECT_BY_GNAME)
+    @command("q", ["--verbose"], readonly=True)
     def status(self, argv):
-        self.export(argv, statussing=True)
+        options = {
+            'verbose': False,
+            'statussing': True,
+        }
+
+        while len(argv) > 0 and argv[0].startswith("-"):
+            opt = argv.pop(0)
+            if opt in ["--verbose", "-v"]:
+                options['verbose'] = True
+            else:
+                raise ArgumentException("No such option: %s" % opt)
+        self.raise_unrecognized_argument(argv)
 
-    @command("l", "--concluded --verbose --all", SELECT_BY_GNAME)
+        self.exporter = import_module('dit.ditexporter')
+        self.exporter.setup(sys.stdout, options)
+
+        self.exporter.begin()
+
+        (group, subgroup, task) = self._get_current_task()
+        self._export_task(group, subgroup, task)
+
+        while not self._previous_empty():
+            (group, subgroup, task) = self._previous_pop()
```

Comment:
Although the stack is not being written to disk, this leads to an inconsistent state after leaving the status function. To avoid this, create a copy of previous before the while loop and use the copy inside the loop.

```
_previous_clone():
    return list(self._previous_stack)

p = self._previous_clone()
while len(p) > 0:
    (group, subgroup, task) = p.pop()
    self._export_task(group, subgroup, task)
```